iceoryx_examples/automotive_soa/include/owl/kom/field_subscriber.inl

```diff
@@ -0,0 +1,173 @@
+{
+    IOX_DISCARD_RESULT(maxNumberOfSamples);
+
+    core::Result<size_t> numberOfSamples{0};
+
+    while (m_subscriber.take()
+               .and_then([&](const auto& sample) {
+                   callable(sample.get());
+                   numberOfSamples++;
+               })
+               .or_else([](auto& result) {
+                   if (result != iox::popo::ChunkReceiveResult::NO_CHUNK_AVAILABLE)
+                   {
+                       std::cerr << "Error receiving chunk!" << std::endl;
+                   }
+               }))
+    {
+    }
+    return numberOfSamples;
+}
+
+template <typename T>
+inline Future<T> FieldSubscriber<T>::Get()
+{
+    // If we call Get() twice shortly after each other, once the response of the first request has not yet
+    // arrived, we have a problem
+    //! [FieldSubscriber get]
+    bool requestSuccessfullySend{false};
+    m_client.loan()
+        .and_then([&](auto& request) {
+            request.getRequestHeader().setSequenceId(m_sequenceId);
+            request.send().and_then([&]() { requestSuccessfullySend = true; }).or_else([](auto& error) {
+                std::cerr << "Could not send Request! Error: " << error << std::endl;
+            });
+        })
+        .or_else([&](auto& error) {
+            std::cerr << "Could not allocate Request! Error: " << error << std::endl;
+            requestSuccessfullySend = false;
+        });
+
+    if (!requestSuccessfullySend)
+    {
+        return Future<FieldType>();
+    }
+    return receiveResponse();
+    //! [FieldSubscriber get]
+}
+
+template <typename T>
+inline Future<T> FieldSubscriber<T>::Set(const FieldType& value)
+{
+    // If we call Set() twice shortly after each other, once the response of the first request has not yet
+    // arrived, we have a problem
+    bool requestSuccessfullySend{false};
+    m_client
+        .loan()
+        //! [FieldSubscriber set]
+        .and_then([&](auto& request) {
+            request.getRequestHeader().setSequenceId(m_sequenceId);
+            request->emplace(value);
+            //! [FieldSubscriber set]
+            request.send().and_then([&]() { requestSuccessfullySend = true; }).or_else([](auto& error) {
+                std::cerr << "Could not send Request! Error: " << error << std::endl;
+            });
+        })
+        .or_else([&](auto& error) {
+            std::cerr << "Could not allocate Request! Error: " << error << std::endl;
+            requestSuccessfullySend = false;
+        });
+
+    if (!requestSuccessfullySend)
+    {
+        return Future<FieldType>();
+    }
+    return receiveResponse();
+}
+
+template <typename T>
+inline Future<T> FieldSubscriber<T>::receiveResponse()
+{
+    Promise<FieldType> promise;
+    auto future = promise.get_future();
+    // Typically you would e.g. use a worker pool here, for simplicity we use a plain thread
+    std::thread(
+        [&](Promise<FieldType>&& promise) {
```

Comment:
Actually not a case for `Promise<FieldType>&& promise` but `Promise<FieldType> promise` (and move, as below). However it behaves similarly since the thread decays it to this anyway and invokes the move ctor itself.

Without a thread no move ctor would be invoked with this signature. It is rather subtle though and I am fine with both, but would tend to pass b value since what happens more obvious.